migration/v0.36/prepare/10-thanos-distrib-memory-limit.sh

```diff
@@ -0,0 +1,21 @@
+#!/usr/bin/env bash
+
+HERE="$(dirname "$(readlink -f "${0}")")"
+ROOT="$(readlink -f "${HERE}/../../../")"
+
+# shellcheck source=scripts/migration/lib.sh
+source "${ROOT}/scripts/migration/lib.sh"
+
+if [[ "${CK8S_CLUSTER}" =~ ^(sc|both)$ ]]; then
+  log_info "operation on service cluster"
+  if ! yq_null sc .thanos.receiveDistributor.resources.limits.memory; then
+    log_info "- check if thanos distributor memory limit is less than 1Gi"
+    mblimit=$(yq4 '.thanos.receiveDistributor.resources.limits.memory | select(. == "*Mi") | sub("Mi","")' "${CK8S_CONFIG_PATH}/sc-config.yaml")
```

Comment:
*Suggestion*: I just found the cmd `numfmt` that can convert to bytes

```suggestion
    bytelimit=$(yq4 '.thanos.receiveDistributor.resources.limits.memory' "${CK8S_CONFIG_PATH}/sc-config.yaml" | numfmt --from=auto)
```

Alternatively if we want to be a bit more explicit in our format input:

```suggestion
    bytelimit=$(yq4 '.thanos.receiveDistributor.resources.limits.memory' "${CK8S_CONFIG_PATH}/sc-config.yaml" | numfmt --from=iec-i)
```

That should make this script a bit more stable.

I guess this might be a bit late to change here, so I'm fine with leaving it like this for now and to just try to use this in the future. So I'll let you take that decision.

Reply:
I did a commit using `numfmt`, seems that it didn't like values with a `.`

Reply:
Seems to be a locale issue, because this works

```
$ echo 1.1Gi | LC_ALL=C numfmt --from=auto
1181116007
```

Reply:
Or with my local locale 

```
$ echo 1,1Gi | numfmt --from=auto
1181116007
```